src/Halogen/VDom/Types.purs

```diff
@@ -72,14 +74,21 @@ runGraft =
     in
       go v
 
-data ElemSpec a = ElemSpec (Maybe Namespace) String a
+data ElemSpec a = ElemSpec (Maybe Namespace) ElemName a
 
 instance functorElemSpec ∷ Functor ElemSpec where
   map f (ElemSpec ns name a) = ElemSpec ns name (f a)
 
-newtype Namespace = Namespace String
+newtype ElemName = ElemName String
+
+derive instance newtypeElemName ∷ Newtype ElemName _
+derive newtype instance eqElemName ∷ Eq ElemName
+derive newtype instance ordElemName ∷ Ord ElemName
+derive instance genericElemName ∷ Generic ElemName
```

Comment:
Do you think Generic instances are useful? Since the VDom isn't generic, I don't know what you could do with this.